.github/workflows/e2e-tests.yml

```diff
@@ -317,84 +317,127 @@ jobs:
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-max-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-max-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-open-end-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version-v5
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version-v7
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-no-versions
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config-invalid
               OUTPUT=$(../bashunit -a exit_code "1" ../../bin/phpstan)
               echo "$OUTPUT"
               ../bashunit -a contains 'Invalid configuration' "$OUTPUT"
               ../bashunit -a contains 'Invalid PHP version range: phpVersion.max should be greater or equal to phpVersion.min.' "$OUTPUT"
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config-patch
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
+          - script: |
+              cd e2e/bug-12972
+              OUTPUT=$(../bashunit -a exit_code "1" ../../bin/phpstan)
+              echo "$OUTPUT"
+              ../bashunit -a contains 'Internal error: Failed opening required' "$OUTPUT"
+              ../bashunit -a contains 'e2e/bug-12972/src/OTHER/file.php' "$OUTPUT"
+            os: "ubuntu-latest"
+          - script: |
+              cd e2e/bug-12972
+              ../../bin/phpstan analyze
+            os: "macos-latest"
+          - script: |
+              cd e2e/bug-12972
+              ../../bin/phpstan analyze
+            os: "windows-latest"
```

Comment:
I don't get it. Why do you expect different outcomes on Linux vs. the rest?

Reply:
because linux does not have a case-insensitive filesystem

Reply:
So you have a project that deliberately works only on case-insensitive filesystems?

Reply:
the test mimics what php-src will do at runtime.
it will happily accept a different-cased path on windows/macos but doesn't on linux.

Reply:
I'd like if the e2e test mimiced how your project actually worked. This just looks like PHPStan fixes macOS but breaks Linux.

Reply:
ok, lets look into https://github.com/phpstan/phpstan-src/pull/4003 which mimics our project setup more closely